src/platforms/web/compiler/directives/model.js

```diff
@@ -118,7 +118,7 @@ function genSelect (
     `.map(function(o){var val = "_value" in o ? o._value : o.value;` +
     `return ${number ? '_n(val)' : 'val'}})`
 
-  const assignment = '$event.target.multiple ? $$selectedVal : $$selectedVal[0]'
```

Comment:
$$selectedVal may be an empty array, this way we should not return its first element
reverse the model's previous value is a reasonable chose